opensearch-project/OpenSearch-Dashboards
# [Multiple Datasource] Add multi selectable data source component

### Description

This change adds a new data source component which allows multi selections of data sources. 

### Issues Resolved
Fixes https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6210

## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/17714150/72c8594e-32ed-4437-8dd5-3728ae4f9f94


## Testing the changes
The following steps were performed in the recording above:
1. enable data source plugin
2. go to query workbench plugin page, should see the multi selectable component rendered correctly with all options selected by default, with the badge icon show the total number of data sources
3. click on `deselect all` will deselect all options and the badge icon should show 0
4. select a few data sources, the badge icon should update with number of selected data sources

### Check List
